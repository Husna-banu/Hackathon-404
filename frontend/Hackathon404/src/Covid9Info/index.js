import React, { useState, useEffect } from 'react';
import {
    View,
    Text,
    StatusBar,
    SafeAreaView,
    FlatList,
    TouchableOpacity,
} from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome5';
import styles from './style';
import commonStyle from '../commonStyles';
import orderStyle from '../Order/style';
import Logout from '../components/Logout';

export default function Covid9Info({ route, navigation }) {
    const [stateData, setStateData] = useState({
        hotelName: '',
        hotelId: '',
        listOfServices: '',
    });
    useEffect(() => {
        const { hotelId, listOfServices } = route.params;
        setStateData(state => ({
            ...state,
            hotelId: hotelId,
            listOfServices: listOfServices,
        }));
    }, [route.params]);
    const backToPage = () => {
        navigation.goBack();
    };
    const serviceDetails = (serviceId, serviceName) => {
        if (serviceId && serviceName) {
            navigation.navigate('ServiceDetails', {
                serviceId: serviceId,
                serviceName: serviceName,
            });
        }
    };
    const renderItem = ({ item }) => {
        return (
            <TouchableOpacity
                onPress={() => serviceDetails(item.serviceId, item.serviceName)}>
                <View style={styles.servicesListStyle}>
                    <Text style={styles.serviceNameStyle, commonStyle.fnBold}>{item.serviceName}</Text>
                    <Text style={styles.serviceNameStyle}>Capacity in {item.serviceName} are {item.capacity}</Text>
                    <Text style={styles.serviceNameStyle}>Current Availability of seats  in {item.serviceName} are {item.availableSpace}</Text>
                </View>
            </TouchableOpacity>
        );
    };
    const goToService = () => {
        navigation.navigate('Dashboard', { hotelId: stateData.hotelId, listOfServices: stateData.listOfServices });
    }
    return (
        <View style={commonStyle.container, commonStyle.bgColor}>
            <StatusBar barStyle="dark-content" />

            <SafeAreaView>
                <View style={commonStyle.covidhHeader}>
                    <Text style={commonStyle.heading}>
                        <Text style={{ fontSize: 12, height: 80 }} > The safety and wellbeing of our guests and associates is of paramount importance to Raj Hotel.
                        Associates at Raj Hotel are extensively trained on safety, hygiene and physical distancing norms to ensure guests' wellbeing and delight, every time.
                        Technology is used for seamless, yet personalized check-ins, check-outs and dining experiences. With digital payments and non-invasive thermal screening, the touch of care replaces the physical one. </Text></Text>
                    <Logout navigation={navigation} />
                </View>
                <View style={commonStyle.header}>
                    <Icon
                        name="arrow-left"
                        style={commonStyle.backButton}
                        size={20}
                        onPress={backToPage}
                    />

                    <Text style={commonStyle.heading}>Welcome to Raj Hotel</Text>
                    <Logout navigation={navigation} />
                </View>
                <View style={commonStyle.content}>
                    <FlatList
                        data={stateData.listOfServices}
                        renderItem={renderItem}
                        keyExtractor={item => item.serviceId.toString()}
                    />
                </View>
                <View>
                    <TouchableOpacity>
                        <Text style={orderStyle.textInput} onPress={goToService}>
                            Go to Service List
              </Text>
                    </TouchableOpacity>
                </View>
            </SafeAreaView>
        </View>
    );
}
