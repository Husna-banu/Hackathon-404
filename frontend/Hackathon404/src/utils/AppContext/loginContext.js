import React from 'react'
const accessStateContext = React.createContext()
const accessDispatchContext = React.createContext()

function AccessReducer(state, action) {
  switch (action.type) {
    case 'login': {
      return { loginName: action.payload, login: true }
    }
    case 'logout': {
      return { loginName: action.payload, login: false }
    }
    default: {
      throw new Error(`Unhandled action type: ${action.type}`)
    }
  }
}

function AccessProvider({ children }) {
  const [state, dispatch] = React.useReducer(AccessReducer, { loginName: '', login: false })
  return (
    <accessStateContext.Provider value={state}>
      <accessDispatchContext.Provider value={dispatch}>
        {children}
      </accessDispatchContext.Provider>
    </accessStateContext.Provider>
  )
}

function useAccessState() {
  const context = React.useContext(accessStateContext)
  if (context === undefined) {
    throw new Error('useAccessState must be used within a AccessProvider')
  }
  return context
}

function useAccessDispatch() {
  const context = React.useContext(accessDispatchContext)
  if (context === undefined) {
    throw new Error('useAccessDispatch must be used within a AccessProvider')
  }
  return context
}

export { AccessProvider, useAccessState, useAccessDispatch }
