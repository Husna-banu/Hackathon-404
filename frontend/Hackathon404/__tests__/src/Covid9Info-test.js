import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Covid9Info from '../../src/Covid9Info';
import {AccessProvider} from '../../src/utils/AppContext/loginContext';

jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper');
fetch = jest.fn(() => Promise.resolve());
const props = {
  navigate: jest.fn(),
  route: {
    params: {},
  },
};

it('renders correctly', async () => {
  const container = renderer.create(
    <AccessProvider>
      <Covid9Info {...props} />
    </AccessProvider>,
  );
  expect(container).toMatchSnapshot();
});
