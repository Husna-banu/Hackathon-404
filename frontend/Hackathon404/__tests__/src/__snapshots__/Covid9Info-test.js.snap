// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#CDE6FA",
      "height": "100 %",
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "backgroundColor": "yellow",
          "flexDirection": "row",
          "height": 150,
          "marginBottom": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "alignItems": "center",
            "fontSize": 20,
            "fontWeight": "bold",
            "justifyContent": "center",
            "paddingLeft": 10,
            "textAlign": "center",
          }
        }
      >
        <Text>
           
          The safety and wellbeing of our guests and associates is of paramount importance to Raj Hotel. Associates at Raj Hotel are extensively trained on safety, hygiene and physical distancing norms to ensure guests' wellbeing and delight, every time. Technology is used for seamless, yet personalized check-ins, check-outs and dining experiences. With digital payments and non-invasive thermal screening, the touch of care replaces the physical one.
           
        </Text>
      </Text>
    </View>
    <View
      style={
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "backgroundColor": "skyblue",
          "flexDirection": "row",
          "height": 50,
        }
      }
    >
      <Text
        allowFontScaling={false}
        onPress={[Function]}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 20,
            },
            Object {
              "fontSize": 20,
              "paddingLeft": 10,
              "paddingRight": 20,
            },
            Object {
              "fontFamily": "FontAwesome5Free-Solid",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {
              "fontWeight": "900",
            },
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Object {
            "alignItems": "center",
            "fontSize": 20,
            "fontWeight": "bold",
            "justifyContent": "center",
            "paddingLeft": 10,
            "textAlign": "center",
          }
        }
      >
        Welcome to Raj Hotel
      </Text>
      <Text
        allowFontScaling={false}
        onPress={[Function]}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 20,
            },
            Object {
              "position": "absolute",
              "right": 15,
            },
            Object {
              "fontFamily": "anticon",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "marginBottom": 20,
          "padding": 10,
        }
      }
    >
      <RCTScrollView
        data=""
        disableVirtualization={false}
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={false}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        onContentSizeChange={[Function]}
        onEndReachedThreshold={2}
        onLayout={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        stickyHeaderIndices={Array []}
        updateCellsBatchingPeriod={50}
        viewabilityConfigCallbackPairs={Array []}
        windowSize={21}
      >
        <View />
      </RCTScrollView>
    </View>
  </RCTSafeAreaView>
</View>
`;
